integration/kubernetes/init.sh

```diff
@@ -49,18 +41,30 @@ echo "Start ${cri_runtime} service"
 sudo systemctl start ${cri_runtime}
 
 echo "Init cluster using ${cri_runtime_socket}"
-sudo -E kubeadm init --pod-network-cidr 10.244.0.0/16 --cri-socket="unix://${cri_runtime_socket}"
+kubeadm_config_template="${SCRIPT_PATH}/kubeadm/config.yaml"
+kubeadm_config_file="$(mktemp --tmpdir kubeadm_config.XXXXXX.yaml)"
+
+sed -e "s|CRI_RUNTIME_SOCKET|${cri_runtime_socket}|" "${kubeadm_config_template}" > "${kubeadm_config_file}"
+
+sudo -E kubeadm init --config "${kubeadm_config_file}"
+
 export KUBECONFIG=/etc/kubernetes/admin.conf
 
 sudo -E kubectl get nodes
 sudo -E kubectl get pods
-sudo -E kubectl create -f "${SCRIPT_PATH}/data/kube-flannel-rbac.yml"
-sudo -E kubectl create --namespace kube-system -f "${SCRIPT_PATH}/data/kube-flannel.yml"
 
-# The kube-dns pod usually takes around 30 seconds to get ready
+# kube-flannel config file taken from k8s 1.12 documentation:
+flannel_config="https://raw.githubusercontent.com/coreos/flannel/bc79dd1505b0c8681ece4de4c0d86c5cd2643275/Documentation/kube-flannel.yml"
+
+sudo -E kubectl apply -f "$flannel_config"
+
+# The kube-dns pod usually takes around 120 seconds to get ready
 # This instruction will wait until it is up and running, so we can
 # start creating our containers.
-dns_wait_time=300
+dns_wait_time=120
```

Comment:
I'm not really clear on this change - if it takes ~2 minutes to start, why are we reducing the time we wait in case it takes a little longer?

Reply:
well, it is actually taking less than 2 minutes, so I wanted to try reducing the time from 300s. So far it has been working well with the 2 minutes timeout.